scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -241,15 +241,35 @@ def test_best_solution_retrieval(self):
     def test_callback_terminates(self):
         # test that if the callback returns true, then the minimization halts
         bounds = [(0, 2), (0, 2)]
-
-        def callback(param, convergence=0.):
+        expected_msg = 'callback function requested stop early by returning True'
+
+        def callback_numpy_true(param, convergence=0.):
+            # This is an odd way to return true from a callback,
+            # but doing it this way exercises an important piece
+            # of functionality. It's important that the callback
+            # be able to halt execution  when returning the True
+            # that would be produced by a numpy array comparison
+            # which differs in the `is`  sense from the one that
+            # python uses. 
+            #  ie, (np.array([1]) == 1)[0] is True => False
+            #      (np.array([1]) == 1)[0] == True => True
+            return (np.array([1]) == 1)[0]
```

Comment:
This test is redundant because the callback returns the same as `callback_python_true`:

```
>>> (np.array([1]) == 1)[0]
True
```

Reply:
You already test for `bool(val)` by `callback_evaluates_true` anyway.